### ISSUE TITLE
cmd/go: vendor experiment and running tests on ./...

### ISSUE THREAD

**Issue by pkieltyka** [golang/go#12239]
```shell
$ cd $GOPATH/myproject
$ GO15VENDOREXPERIMENT=1 go test ./...
```

will run the tests in `./vendor` since the `./...` will expand there as well, I believe `./vendor` should be skipped in that instance

**Comment by minux** [comment]
Merged into #11659.

**Comment by arekkas** [comment]
In go 1.6 the same error occurs when `GO15VENDOREXPERIMENT=1` is not set - but `GO15VENDOREXPERIMEN` is enabled by default so it should not need to be set.
